### PR TITLE
test: port `recycle` tests from Stunt

### PR DESCRIPTION
**Pull request by abesto**
* Verified this passes against MOO-1.8.1
* We have some failures against `moor`. First one:

```
assertion failed: `(left == right)`: Line 18: recycle($nothing);

Diff < left / right > :
<E_INVIND
>E_INVARG
```

### CODE REVIEW

File: crates/moot/tests/moot_lmoo.rs [hunk 1]
```diff
@@ -84,7 +84,7 @@ impl TelnetMootRunner {
     fn resolve_response(&mut self, response: String) -> Result<String, std::io::Error> {
         // Resolve the response; for example, the test assertion may be `$object`; resolve it to the object's specific number.
         self.client(WIZARD).command(format!(
-            "; {response}; \"TelnetMootRunner::resolve_response\";"
```
Comment (abesto) [inline crates/moot/tests/moot_lmoo.rs:87]: Yeah this was completely busted, without the `return` we just... accepted everything :D

**Comment by rdaum** [comment]
E_INVIND I guess is translated from `ObjectNotFound`, so hm. I wonder.

I'll take a closer look tomorrow

**Comment by abesto** [comment]
Oh yeah of course. I'd already fixed this in the experimental branch I threw away at https://github.com/rdaum/moor/pull/328/commits/4ff11fc10c8414901c2097a442950eb641557f1f#diff-ad02df21006e376aae81ce146139a598ce81d4c03eb9372853428945c0993a02R202. Lemme bring that change back in.

File: crates/kernel/src/builtins/bf_properties.rs [hunk 2]
```diff
@@ -204,7 +204,7 @@ fn bf_add_property(bf_args: &mut BfCallState<'_>) -> Result<BfRet, BfErr> {
             *location,
             *location,
             name.as_str(),
-            bf_args.caller_perms(),
+            attrs.owner.unwrap(),
```
Comment (abesto) [inline crates/kernel/src/builtins/bf_properties.rs:207]: This kind of stuff will be more fully tested when I port https://github.com/toddsundsted/stunt/blob/master/test/test_objects_and_properties.rb; this is enough to make the `recycle` suite pass.

**Pull request merged**